src/gic_v2.rs

```diff
@@ -465,3 +513,126 @@ impl GicCpuInterface {
+        }
+    }
+
+    /// Get a reference to the GIC Hypervisor Interface Registers.
+    ///
+    /// # Safety
+    ///
+    /// This function dereferences a raw pointer, so it is marked as unsafe.
+    const fn regs(&self) -> &GicHypervisorInterfaceRegs {
+        unsafe { self.base.as_ref() }
+    }
+
+    /// Get the value of the HCR (Hypervisor Control Register).
+    ///
+    /// # Returns
+    ///
+    /// The current value of the HCR register.
+    pub fn get_hcr(&self) -> u32 {
+        self.regs().HCR.get()
+    }
+
+    /// Set the value of the HCR (Hypervisor Control Register).
+    ///
+    /// # Arguments
+    ///
+    /// * `hcr` - The value to set the HCR register to.
+    pub fn set_hcr(&self, hcr: u32) {
+        self.regs().HCR.set(hcr);
+    }
+
+    /// Get the value of the ELRSR (Empty List Register Status Register) at the specified index.
+    ///
+    /// # Arguments
+    ///
+    /// * `idx` - The index of the ELRSR register to read.
+    ///
+    /// # Returns
+    ///
+    /// The current value of the specified ELRSR register.
+    pub fn elrsr(&self, idx: usize) -> u32 {
+        self.regs().ELRSR[idx].get()
+    }
+
+    /// Get the value of the EISR (Empty Interrupt Status Register) at the specified index.
+    ///
+    /// # Arguments
+    ///
+    /// * `idx` - The index of the EISR register to read.
+    ///
+    /// # Returns
+    ///
+    /// The current value of the specified EISR register.
+    pub fn eisr(&self, idx: usize) {
+        self.regs().EISR[idx].get();
+    }
+
+    /// Get the value of the MISR (Maintenance Interrupt Status Register).
+    ///
+    /// # Returns
+    ///
+    /// The current value of the MISR register.
+    pub fn misr(&self) -> u32 {
+        self.regs().MISR.get()
+    }
+
+    /// Get the value of the APR (Active Priority Register).
+    ///
+    /// # Returns
+    ///
+    /// The current value of the APR register.
+    pub fn apr(&self) -> u32 {
+        self.regs().APR.get()
+    }
+
+    /// Get the value of the LR (List Register) at the specified index.
+    ///
+    /// # Arguments
+    ///
+    /// * `idx` - The index of the LR register to read.
+    ///
+    /// # Returns
+    ///
+    /// The current value of the specified LR register.
+    pub fn get_lr(&self, idx: usize) -> u32 {
+        self.regs().LR[idx].get()
+    }
+
+    /// Set the value of the LR (List Register) at the specified index.
+    ///
+    /// # Arguments
+    ///
+    /// * `idx` - The index of the LR register to write.
+    /// * `lr` - The value to set the LR register to.
+    pub fn set_lr(&self, idx: usize, lr: u32) {
+        self.regs().LR[idx].set(lr);
+    }
+
+    /// Get the value of the VTR (Virtualization Translation Register).
+    ///
+    /// # Returns
+    ///
+    /// The number of List Registers supported by the GIC, derived from the VTR register.
+    pub fn get_vtr(&self) -> u32 {
+        let vtr: u32 = self.regs().VTR.get();
```

Comment:
refactor `VTR` like https://github.com/arceos-hypervisor/arm_gicv2/blob/main/src/regs/gicd_sgir.rs

for better readability

Reply:
ok